golioth/pytest/golioth_test.py

```diff
@@ -0,0 +1,87 @@
+import os
+import pytest
+from golioth import Client, Project, Blueprint, Tag
+
+proj_name = 'firmware_ci'
+tag_name = 'pytest'
+bp_name = 'pytest_bp'
+
+api_key = os.environ['GOLIOTH_API_KEY']
+client = Client(api_url = "https://api.golioth.io", api_key = api_key)
+ci_proj = Project(client, {'id':None, 'name':None})
+
+
+@pytest.mark.asyncio
+async def test_projects():
+    global ci_proj
+
+    proj_list = await client.get_projects()
+
+    for p in proj_list:
+        if p.name == proj_name:
+            ci_proj = p
+
+    assert ci_proj.name == proj_name
+
+    proj_by_name = await client.project_by_name(proj_name)
+    assert proj_by_name.id == ci_proj.id
+
+
+@pytest.mark.asyncio
+async def test_tags():
```

Comment:
It would be good to break these test functions up into smaller tests, so each test only tests one specific piece of functionality.

Reply:
So combining this with my comment above, I think we should have one test file per class, and that file should have multiple test functions within it. So for example, `test_tags.py` might have `test_create()`, `test_get_by_name()`, `test_delete()`, etc.

Reply:
Great, thanks for the points on this! I've made the suggested changes. Let me know if I've done anything odd with the tests :smiley: